src/main/java/org/arl/fjage/param/NamedParameter.java

```diff
@@ -74,20 +74,22 @@ public String toString() {
   }
 
   /**
-   * Compares if two named parameters are equals.
+   * Compares if a parameter is equivalent to the named parameter.
    *
    * @return true if two parameters are equal
    */
   @Override
   public boolean equals(Object obj) {
-    if (!(obj instanceof NamedParameter)) return false;
-    NamedParameter p = (NamedParameter)obj;
+    if (!(obj instanceof Parameter)) return false;
     String name1 = name;
     if (name1.contains(".")) name1 = name1.substring(name1.indexOf('.')+1);
-    String name2 = p.name;
+    String name2 = obj.toString();
     if (name2.contains(".")) name2 = name2.substring(name2.indexOf('.')+1);
     if (!name1.equals(name2)) return false;
```

Comment:
I assume the intention here is that this line compares unqualified names. However, that would mean that e.g. `NodeInfoParam.address` and `CellMobileParam.address` would compare equal.

Reply:
No, I don't believe the intent here was for unqualified names. I don't recall the exact reason why this peeling off of the first segment was introduced (and I tried going back to history to reconstruct, but failed at understanding it), but I vaguely remember it had something to do with how some of the gateways dealt with parameters.

Non-named parameters are always compared fully qualified.

Reply:
I believe currently the behaviour is like this:

- [x] Enum item to enum item is compared using the full path. 
- [ ] Enum item to `NamedParameter` is always `false`.
- [ ] Fully qualified `NamedParameter` to enum item is "mostly" `false`. (Comparison is done based on `toString()`, and enums' `toString()` method returns only the item name but not the path. The "mostly" is because currently we only peel off the first component of the path.)
- [x] Unqualified `NamedParameter` to enum item is compared based on only the name. 
- [ ] `NamedParameter` to `NamedParameter` is compared based on all but the first component of the path. 

The ticks indicate behaviour as I would expect it.